# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: layout/svg/SVGIntegrationUtils.cpp
# Commit: 63f54f50c32d
# Full Hash: 63f54f50c32dcbfa01321485c0cafad0c5bbfbba
# Author: Jeff Muizelaar <jmuizelaar@mozilla.com>
# Date: 2022-02-19 09:33:23
# Regressor Bug: 1756197
# File Overlap Count: 1
# Description:
#   Bug 1756197. Avoid clipping if we're just going to fill the path. r=gfx-reviewers,nical
#   
#   This avoids having to rasterize the path twice. Once for the clip path and once for
#   the fill. It will also give us correct antialiasing on the edges.
#   
# ==============================================================================

diff -r 0886d48d6a83 -r 63f54f50c32d layout/svg/SVGIntegrationUtils.cpp
--- a/layout/svg/SVGIntegrationUtils.cpp	Fri Feb 18 22:04:43 2022 +0000
+++ b/layout/svg/SVGIntegrationUtils.cpp	Fri Feb 18 22:04:43 2022 +0000
@@ -758,16 +758,19 @@
     MoveContextOriginToUserSpace(firstFrame, aParams);
 
     basicShapeSR.SetContext(&ctx);
-    CSSClipPathInstance::ApplyBasicShapeOrPathClip(
-        ctx, frame, SVGUtils::GetCSSPxToDevPxMatrix(frame));
+    gfxMatrix mat = SVGUtils::GetCSSPxToDevPxMatrix(frame);
     if (!maskUsage.shouldGenerateMaskLayer) {
       // Only have basic-shape clip-path effect. Fill clipped region by
       // opaque white.
       ctx.SetDeviceColor(DeviceColor::MaskOpaqueWhite());
+      RefPtr<Path> path = CSSClipPathInstance::CreateClipPathForFrame(
+          ctx.GetDrawTarget(), frame, mat);
+      ctx.SetPath(path);
       ctx.Fill();
 
       return true;
     }
+    CSSClipPathInstance::ApplyBasicShapeOrPathClip(ctx, frame, mat);
   }
 
   // Paint mask onto ctx.